docs/nitro/outcomes.md

```diff
@@ -0,0 +1,49 @@
+
+Nitro supports multiple different assets (e.g. ETH and one or more ERC20s) being held in the same channel.
+
+An outcome is a set of **asset outcomes**.
+
+```
+Outcome = AssetOutcome[]
+AssetOutome = (AssetID, AllocationOrGuarantee)
+AllocationOrGuarantee = Allocation | Guarantee
+Allocation = AllocationItem[]
+AllocationItem = (Destination, Amount)
+Guarantee = (ChannelAddress, Destination[])
+Destination = ChannelAddress | ExternalAddress
+```
+
+Destinations are
+
+An allocation determines
+
+## Implementation
+
+Nitro Adjudicator extends OptimizedForceMove
+Adds a `pushOutcome`
+
+One adjudicator pushing to multiple asset holders.
+
+### Formats
+
+| **Field**             | **Data type**          | **Definition / Explanation**                                                                                                                                  |
+| :-------------------- | :--------------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| Outcome               | `bytes`                | Abi-encoded array of AssetOutcomes                                                                                                                            |
+| AssetOutcome          | `(address, bytes)`     | `(assetHolderAddress, data)`, where the asset holder is the contract which holds the given type of asset, and `data` is the abi-encoded AllocationOrGuarantee |
+| AllocationOrGuarantee | `(uint8, bytes)`       | `(type, data)`, where `type` is 0 for allocations and 1 for guarantees, and `data` field is the abi-encoded form of the Allocation or Guarantee               |
+| Allocation            | `(bytes32, uint256)[]` | An list of AllocationItems, in payout priority order (highest priority first)                                                                                 |
+| AllocationItem        | `(bytes32, uint256)`   | `(destination, amount)`                                                                                                                                       |
+| Guarantee             | `(bytes32, bytes32[])` | `(guaranteedChannelAddress, destinations)`                                                                                                                    |
+| ChannelAddress        | `bytes32`              | Equal to the ChannelId defined in ForceMove                                                                                                                   |
+| ExternalAddress       | `bytes32`              | An Ethereum address left padded with 12 bytes of zeros                                                                                                        |
+| Destination           | `bytes32`              | Taken to be an ExternalAddress if starts with 12 bytes of leading zeros and a ChannelAddress otherwise                                                        |
```

Comment:
It would be more convenient if an ExternalAddress *ends* with 12 bytes of *trailing* zeros. This would allow a particularly neat implementation of helper functions such as  https://github.com/statechannels/nitro-protocol/blob/6d2064c7aecb62fb36442e7db504e0f0ce0b2a86/contracts/AssetHolder.sol#L44-L46

Reply:
This implementation passes the following tests:

https://github.com/statechannels/nitro-protocol/blob/6d2064c7aecb62fb36442e7db504e0f0ce0b2a86/test/ETHAssetHolder/isExternalAddress.test.ts#L24-L35

Reply:
Looks great - will change.